lib/Backends/Habana/HabanaDeviceManager.cpp

```diff
@@ -114,8 +114,15 @@ llvm::Error HabanaDeviceManager::init() {
 llvm::Error HabanaDeviceManager::updateMemoryUsage() {
   // TODO: Use synGetMemInfo once implemented.
 
-  totalMemory_ = uint64_t{GlowHabanaMemory} * 1024;
-  freeMemory_ = uint64_t{GlowHabanaMemory} * 1024;
+  // Use GlowHabanaMemory if it is defined from GFLAGS or llvm params,
+  // otherwise, fall back to what config says.
+  uint64_t defaultMemory = 7 << 20;
+  if (GlowHabanaMemory == defaultMemory && config_.getDeviceMemory() != 0) {
+    totalMemory_ = config_.getDeviceMemory();
```

Comment:
how does it work if `getDeviceMemory()` method expects a param passed in.

Reply:
> how does it work if `getDeviceMemory()` method expects a param passed in.

I defined two getDeviceMemory(). one with param(a default value ) and one without :)